### ISSUE TITLE
Normal Loss

### ISSUE THREAD

**Issue by SuperXia123** [chengweialan/DeSiRe-GS#4]
Thanks for your great work. May I ask why the normal loss is commented out here? In my opinion this loss is the key that garantees geometry regularization according to the paper.
![111](https://github.com/user-attachments/assets/f35f4b5c-f899-414f-922a-02eb61956562)

**Comment by sonnefred** [comment]
Same question. Also, in the paper, there is a loss to flatten 3D Gaussians, but I didn't find the relevant codes. Do you know where it is? Thanks.
![图片](https://github.com/user-attachments/assets/9b1384e3-6c1e-4984-864a-bf855f41a7ce)

**Comment by sonnefred** [comment]
> Thanks for your great work. May I ask why the normal loss is commented out here? In my opinion this loss is the key that garantees geometry regularization according to the paper. ![111](https://private-user-images.githubusercontent.com/49649598/391417855-f35f4b5c-f899-414f-922a-02eb61956562.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMxOTU3MjcsIm5iZiI6MTczMzE5NTQyNywicGF0aCI6Ii80OTY0OTU5OC8zOTE0MTc4NTUtZjM1ZjRiNWMtZjg5OS00MTRmLTkyMmEtMDJlYjYxOTU2NTYyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjAzVDAzMTAyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMzYTI2Nzk2MWNmZTQ0NmQ2OGQ3OGFkNTI5YWVjMmY5YTI0ZTYwYTllNDI5NGMxZmNhZWY2YThlNGM3YjA3ZDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.-G3HAaUT1L_WbtsTPzbZ9h3vSuECw-V3s9M6izK3W2Y)

I think maybe we should refer to pvg_render.py, where all loss functions are used.

**Comment by chengweialan** [comment]
You are right. We use `pvg_render.py` as our renderer, and `gs_render.py` is used for ablation only.

You can check the configurations here:
https://github.com/chengweialan/DeSiRe-GS/blob/main/configs/emer_reconstruction_stage2.yaml#L83

**Comment by wangjksjtu** [comment]
Thanks for the nice work! I have the same question, seems the scaling regularization is not used? I only see `lambda_scaling: 0.0` in the config and the loss is also doing different thing (trying to make three dimensions close).

![image](https://github.com/user-attachments/assets/7b0eeaa5-2b8c-4ab0-a323-9f70b996e5f6)

```
    if args.lambda_scaling > 0:
        scaling = render_pkg["scaling"]  # (N, 3)
        scaling_loss = (
            (scaling - scaling.mean(dim=-1, keepdim=True)).abs().sum(-1).mean()
        )
        lambda_scaling = (
            args.lambda_scaling * 0.1
        )  # - 0.99 * args.lambda_scaling * min(1, 4 * iteration / args.iterations)
        log_dict["scaling_loss"] = scaling_loss.item()
        loss = loss + lambda_scaling * scaling_loss
```

**Comment by wangjksjtu** [comment]
nvm, I find the loss here ;)
```
        if args.lambda_min_scale > 0:
            min_scale_loss = torch.relu(sorted_scale[..., 0] - args.min_scale) + torch.relu(args.min_scale - sorted_scale[..., 0]) ** 2 # (N)
            log_dict["min_scale"] = min_scale_loss.mean().item()
            loss += args.lambda_min_scale * min_scale_loss.mean()
```
